source/coap_service_api.c

```diff
@@ -310,10 +310,16 @@ int8_t coap_service_initialize(int8_t interface_id, uint16_t listen_port, uint8_
         return -1;
     }
 
-    if (0 > coap_connection_handler_open_connection(this->conn_handler, listen_port, ((this->service_options & COAP_SERVICE_OPTIONS_EPHEMERAL_PORT) == COAP_SERVICE_OPTIONS_EPHEMERAL_PORT),
-                                              ((this->service_options & COAP_SERVICE_OPTIONS_SECURE) == COAP_SERVICE_OPTIONS_SECURE),
-                                              ((this->service_options & COAP_SERVICE_OPTIONS_VIRTUAL_SOCKET) != COAP_SERVICE_OPTIONS_VIRTUAL_SOCKET),
-                                              ((this->service_options & COAP_SERVICE_OPTIONS_SECURE_BYPASS) == COAP_SERVICE_OPTIONS_SECURE_BYPASS))){
+    if (this->service_options & COAP_SERVICE_OPTIONS_SELECT_SOCKET_IF) {
+        socket_interface_selection = this->interface_id;
+    }
+
+    if (0 > coap_connection_handler_open_connection(this->conn_handler, listen_port,
+            ((this->service_options & COAP_SERVICE_OPTIONS_EPHEMERAL_PORT) == COAP_SERVICE_OPTIONS_EPHEMERAL_PORT),
```

Comment:
While touching this, maybe reduce the wordiness? Just `this->service_options & COAP_SERVICE_OPTIONS_EPHEMERAL_PORT`, `!(this->service_options & COAP_SERVICE_OPTIONS_VIRTUAL_SOCKET)`, etc?

(The assignment to bool_t coerces the bitmask to 1 or 0 for you by having an implicit `!= 0`, like `if`).